daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetOutputter.scala

```diff
@@ -0,0 +1,213 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.infoset
+
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.xml.XMLUtils
+import org.xml.sax.helpers.AttributesImpl
+import org.xml.sax.ContentHandler
+import org.xml.sax.SAXException
+
+class SAXInfosetOutputter(ch: ContentHandler) extends InfosetOutputter {
+  /**
+   * Reset the internal state of this InfosetOutputter. This should be called
+   * inbetween calls to the parse method.
+   */
+  override def reset(): Unit = {
+    //do nothing
+  }
+
+  /**
+   * Called by Daffodil internals to signify the beginning of the infoset.
+   *
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def startDocument(): Boolean = {
+    try {
+      ch.startDocument
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  /**
+   * Called by Daffodil internals to signify the end of the infoset.
+   *
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def endDocument(): Boolean = {
+    try {
+      ch.endDocument()
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  /**
+   * Called by Daffodil internals to signify the beginning of a simple element.
+   *
+   * @param diSimple the simple element that is started. Various fields of
+   *                 DISimple can be accessed to determine things like the
+   *                 value, nil, name, namespace, etc.
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def startSimple(diSimple: DISimple): Boolean = {
+    val ns = if (diSimple.erd.thisElementsNamespace.isNoNamespace) "" else diSimple.erd.thisElementsNamespace.toString
+    val prefix = if (diSimple.erd.thisElementsNamespacePrefix == null) "" else diSimple.erd.thisElementsNamespacePrefix
+    val nsb = diSimple.namedQName
+    try {
+      if (prefix.nonEmpty && ns.nonEmpty) {
+        ch.startPrefixMapping(prefix, ns)
+      }
+
+      if (isNilled(diSimple)) { // TODO split this up into isNilled, hasValue or else (empty)
```

Comment:
Agreed on the TODO, this could probably be split up to avoid duplication, e.g.
```scala
val attrs =
  if (isNilled(diSimple)) {
    // create new AttributesImpl and add xsi:nil
  } else {
    // create empty AttributesImp
  }

ch.starElement(..., attrs)

if (diSimple.hasValue) {
  // ch.characters stuff
}
```